Run all unit tests against python3

kahowell opened the pull request:
Previously we were just running the python-rhsm tests.

I renamed the job and script accordingly.

Note that you can use `sh resources/subscription-manager-python3-tests.sh`
from a subscription-manager checkout to verify the script changes.

kahowell committed:
Run all unit tests against python3

Previously we were just running the python-rhsm tests.

I renamed the job and script accordingly.

Note that you can use `sh resources/subscription-manager-python3-tests.sh`
from a subscription-manager checkout to verify the script changes.

jirihnidek reviewed (changes requested):
I did not have installed virtualenv for Python 2. I had installed only virtualenv-3 for Python 3 and running test script failed for this reason. Please modify line 22 in file: `resources/subscription-manager-python3-tests.sh`

(I solved my problem with creating symbolic link)

FYI: I run test remotely in VM and test became frozen in the middle of one GUI test. I thing this is not big issue due to planed disabling of GUI unit tests.

kahowell commented:
> I did not have installed virtualenv for Python 2. I had installed only virtualenv-3 for Python 3 and running test script failed for this reason. Please modify line 22 in file: resources/subscription-manager-python3-tests.sh

Will follow-up with another commit for this.

> FYI: I run test remotely in VM and test became frozen in the middle of one GUI test. I thing this is not big issue due to planed disabling of GUI unit tests.

Let's make sure candlepin/subscription-manager#1762 is merged first, just in case.

kahowell committed:
Fallback to virtualenv-3

If for whatever reason python-virtualenv is not installed, but
python3-virtualenv is, the script will still work.

jirihnidek reviewed (approved):
ACK
